stellargraph/data/explorer.py

```diff
@@ -107,6 +81,114 @@ def run(self, **kwargs):
         """
         raise NotImplementedError
 
+    def _raise_error(self, msg):
+        full = "({}) " + msg
+        raise ValueError(full.format(type(self).__name__))
+
+    def _check_parameter_values(self, nodes=None, n=0, length=0, seed=None):
```

Comment:
wouldn't it be nicer to define all the internal methods before defining methods like `run`?

Reply:
Styles vary - some people like to define all helper functions before first use (a la Pascal), and some like to define them after first use. I don't think we have a standard yet.